src/main.rs

```diff
@@ -1,28 +1,76 @@
-#![cfg(feature = "bin")]
 #![warn(clippy::pedantic)]
 #![forbid(clippy::unwrap_used)]
 
 use std::env::args;
+use std::ops::Deref;
+use std::sync::Arc;
 
 use anyhow::{self, bail, Context};
+use async_trait::async_trait;
 use tokio::fs;
 use tokio::io::{stdin, AsyncReadExt};
+use tracing::{debug, error, info, trace, warn};
 use tracing_subscriber::prelude::*;
 use wascap::jwt;
-use wasmcloud::capability::{HandlerFunc, HostInvocation};
-use wasmcloud::{Actor, Runtime};
+use wasmcloud_host::capability::{host, logging};
+use wasmcloud_host::{Actor, Runtime};
 
-#[allow(clippy::unused_async)]
-async fn host_call(
+#[derive(Clone)]
+struct Handler {
     claims: jwt::Claims<jwt::Actor>,
-    binding: String,
-    invocation: HostInvocation,
-    _call_context: Option<Vec<u8>>,
-) -> anyhow::Result<Option<[u8; 0]>> {
-    bail!(
-        "cannot execute `{invocation:?}` within binding `{binding}` for actor `{}`",
-        claims.subject
-    )
+}
+
+#[derive(Clone)]
+struct HandlerArc(Arc<Handler>);
```

Comment:
This signature feels a little wacky but it might just be because I haven't seen it

Reply:
If Rust would allow implementing traits on an Arc of a type, it would not be necessary. This should go away once `wit-bindgen`-generated traits don't take a `&mut self` anymore, but I have to first raise a discussion point about it upstream (or maybe just a PR)

Reply:
Thanks! That sounds good 👍🏻